### ISSUE TITLE
slur + chord interaction

### ISSUE THREAD

**Issue by craigsapp** [rism-ch/verovio#775]
Here is a case where a slur collides with a note in a chord.  It seems that the calculation of the position of the slur is based on the bottom note, and the top note is not taken into consideration.  Notice that the placement of the slur endpoint drops if the bottom note of the chord goes down in pitch in the second measure, even though the top note stays the same.

<img width="817" alt="screen shot 2018-02-25 at 7 37 11 pm" src="https://user-images.githubusercontent.com/3487289/36652332-f8127244-1a62-11e8-9403-6bf81f6682cf.png">

Instead, the top not of the chord in this sort of configuration (or maybe the note on the stem side of the chord) should be used in the calculation of the slur placement.

MEI data:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="http://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="http://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="4.0.0">
    <meiHead>
        <fileDesc>
            <titleStmt>
                <title />
            </titleStmt>
            <pubStmt />
        </fileDesc>
        <encodingDesc>
            <appInfo>
                <application isodate="2018-02-25T19:35:04" version="2.0.0-dev-c1b52a3">
                    <name>Verovio</name>
                    <p>Transcoded from Humdrum</p>
                </application>
            </appInfo>
        </encodingDesc>
        <workDesc>
            <work>
                <titleStmt>
                    <title />
                </titleStmt>
            </work>
        </workDesc>
    </meiHead>
    <music>
        <body>
            <mdiv xml:id="mdiv-0000002142725165">
                <score xml:id="score-0000000348299707">
                    <scoreDef xml:id="scoredef-0000002094610747">
                        <staffGrp xml:id="staffgrp-0000000088070230">
                            <staffDef xml:id="staffdef-0000001707971213" clef.shape="G" clef.line="2" n="1" lines="5">
                                <label xml:id="label-0000002094324836" />
                            </staffDef>
                        </staffGrp>
                    </scoreDef>
                    <section xml:id="section-0000001262605439">
                        <measure xml:id="measure-L3">
                            <staff xml:id="staff-L3F1N1" n="1">
                                <layer xml:id="layer-L3F1N1" n="1">
                                    <beam xml:id="beam-L4F1-L5F1">
                                        <note xml:id="note-L4F1" dur="8" oct="5" pname="e" accid.ges="n" />
                                        <note xml:id="note-L5F1" dur="8" oct="5" pname="d" accid.ges="n" />
                                    </beam>
                                    <chord xml:id="chord-L6F1" dur="2">
                                        <note xml:id="note-L6F1S1" oct="4" pname="e" accid.ges="n" />
                                        <note xml:id="note-L6F1S2" oct="5" pname="c" accid.ges="n" />
                                    </chord>
                                </layer>
                            </staff>
                            <slur xml:id="slur-L4F1-L6F1N1" staff="1" startid="#note-L4F1" endid="#chord-L6F1" />
                        </measure>
                        <measure xml:id="measure-L7">
                            <staff xml:id="staff-L7F1N1" n="1">
                                <layer xml:id="layer-L7F1N1" n="1">
                                    <beam xml:id="beam-L8F1-L9F1">
                                        <note xml:id="note-L8F1" dur="8" oct="5" pname="e" accid.ges="n" />
                                        <note xml:id="note-L9F1" dur="8" oct="5" pname="d" accid.ges="n" />
                                    </beam>
                                    <chord xml:id="chord-L10F1" dur="2">
                                        <note xml:id="note-L10F1S1" oct="4" pname="c" accid.ges="n" />
                                        <note xml:id="note-L10F1S2" oct="5" pname="c" accid.ges="n" />
                                    </chord>
                                </layer>
                            </staff>
                            <slur xml:id="slur-L8F1-L10F1N1" staff="1" startid="#note-L8F1" endid="#chord-L10F1" />
                        </measure>
                    </section>
                </score>
            </mdiv>
        </body>
    </music>
</mei>
```

**Comment by craigsapp** [comment]
Here is a related example:

<img width="697" alt="screen shot 2018-06-30 at 11 57 24 am" src="https://user-images.githubusercontent.com/3487289/42128260-3fc444f4-7c5c-11e8-8bc2-5927b7b8d5e0.png">

In the first measure the top slur goes from chord to chord.  In the second measure the top slur goes from top note to top note.  In the third measure is a reference of a single note to single note with a slur connecting them from above.

For the first measure, I would expect the slur to be placed as in the third measure.  If the slur is placed above the chord, then it should connect the top notes of the chord.  Instead it seems to be connected to the bottom notes of the chord.

For the second measure, the starting point of the slur is somewhat reasonable, but the ending is attached to the top of the stem for some reason.

MEI test data:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="http://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="http://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="4.0.0">
    <meiHead>
        <fileDesc>
            <titleStmt>
                <title />
            </titleStmt>
            <pubStmt />
        </fileDesc>
        <encodingDesc>
            <appInfo>
                <application isodate="2018-06-30T11:55:49" version="2.0.0-dev-c25c14a">
                    <name>Verovio</name>
                    <p>Transcoded from Humdrum</p>
                </application>
            </appInfo>
        </encodingDesc>
        <workDesc>
            <work>
                <titleStmt>
                    <title />
                </titleStmt>
            </work>
        </workDesc>
    </meiHead>
    <music>
        <body>
            <mdiv xml:id="mdiv-0000000953881064">
                <score xml:id="score-0000002071534006">
                    <scoreDef xml:id="scoredef-0000001560150015" midi.bpm="400">
                        <staffGrp xml:id="staffgrp-0000001616880723">
                            <staffDef xml:id="staffdef-0000001670575374" clef.shape="G" clef.line="2" meter.count="4" meter.unit="4" n="1" lines="5">
                                <label xml:id="label-0000000785711395" />
                            </staffDef>
                        </staffGrp>
                    </scoreDef>
                    <section xml:id="section-L1F1">
                        <measure xml:id="measure-L3" n="1">
                            <staff xml:id="staff-L3F1N1" n="1">
                                <layer xml:id="layer-L3F1N1" n="1">
                                    <chord xml:id="chord-L4F1" dur="2">
                                        <note xml:id="note-L4F1S1" oct="4" pname="c" accid.ges="n" />
                                        <note xml:id="note-L4F1S2" oct="4" pname="e" accid.ges="n" />
                                        <note xml:id="note-L4F1S3" oct="4" pname="g" accid.ges="n" />
                                        <note xml:id="note-L4F1S4" oct="5" pname="c" accid.ges="n" />
                                    </chord>
                                    <chord xml:id="chord-L5F1" dur="2">
                                        <note xml:id="note-L5F1S1" oct="4" pname="d" accid.ges="n" />
                                        <note xml:id="note-L5F1S2" oct="4" pname="f" accid.ges="n" />
                                        <note xml:id="note-L5F1S3" oct="4" pname="a" accid.ges="n" />
                                        <note xml:id="note-L5F1S4" oct="5" pname="d" accid.ges="n" />
                                    </chord>
                                </layer>
                            </staff>
                            <slur xml:id="slur-L4F1N2-L5F1N1" staff="1" startid="#chord-L4F1" endid="#chord-L5F1" />
                            <slur xml:id="slur-L4F1N1-L5F1N2" staff="1" startid="#chord-L4F1" endid="#chord-L5F1" curvedir="above" />
                        </measure>
                        <measure xml:id="measure-L6">
                            <staff xml:id="staff-L6F1N1" n="1">
                                <layer xml:id="layer-L6F1N1" n="1">
                                    <chord xml:id="chord-L7F1" dur="2">
                                        <note xml:id="note-L7F1S1" oct="4" pname="c" accid.ges="n" />
                                        <note xml:id="note-L7F1S2" oct="4" pname="e" accid.ges="n" />
                                        <note xml:id="note-L7F1S3" oct="4" pname="g" accid.ges="n" />
                                        <note xml:id="note-L7F1S4" oct="5" pname="c" accid.ges="n" />
                                    </chord>
                                    <chord xml:id="chord-L8F1" dur="2">
                                        <note xml:id="note-L8F1S1" oct="4" pname="d" accid.ges="n" />
                                        <note xml:id="note-L8F1S2" oct="4" pname="f" accid.ges="n" />
                                        <note xml:id="note-L8F1S3" oct="4" pname="a" accid.ges="n" />
                                        <note xml:id="note-L8F1S4" oct="5" pname="d" accid.ges="n" />
                                    </chord>
                                </layer>
                            </staff>
                            <slur xml:id="slur-L7F1N2-L8F1N1" staff="1" startid="#chord-L7F1" endid="#chord-L8F1" />
                            <slur xml:id="slur-L7F1N1-L8F1N2" staff="1" startid="#note-L7F1S4" endid="#note-L8F1S4" curvedir="above" />
                        </measure>
                        <measure xml:id="measure-L9" right="end">
                            <staff xml:id="staff-L9F1N1" n="1">
                                <layer xml:id="layer-L9F1N1" n="1">
                                    <note xml:id="note-L10F1" dur="2" oct="5" pname="c" stem.dir="up" accid.ges="n" />
                                    <note xml:id="note-L11F1" dur="2" oct="5" pname="d" stem.dir="up" accid.ges="n" />
                                </layer>
                            </staff>
                            <slur xml:id="slur-L10F1-L11F1N1" staff="1" startid="#note-L10F1" endid="#note-L11F1" curvedir="above" />
                        </measure>
                    </section>
                </score>
            </mdiv>
        </body>
    </music>
</mei>
```

Also, if there are articulations on the stem sides of notes/chords, the slurs must be further away from the noteheads than the articulations:

<img width="545" alt="screen shot 2018-06-30 at 12 05 44 pm" src="https://user-images.githubusercontent.com/3487289/42128316-680a7b30-7c5d-11e8-9668-795c5aac04d7.png">


Particularly in the case of portato, where there are staccatos and slurs:

https://en.wikipedia.org/wiki/Portato

MEI test data:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="http://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="http://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="4.0.0">
    <meiHead>
        <fileDesc>
            <titleStmt>
                <title />
            </titleStmt>
            <pubStmt />
        </fileDesc>
        <encodingDesc>
            <appInfo>
                <application isodate="2018-06-30T12:05:48" version="2.0.0-dev-c25c14a">
                    <name>Verovio</name>
                    <p>Transcoded from Humdrum</p>
                </application>
            </appInfo>
        </encodingDesc>
        <workDesc>
            <work>
                <titleStmt>
                    <title />
                </titleStmt>
            </work>
        </workDesc>
    </meiHead>
    <music>
        <body>
            <mdiv xml:id="mdiv-0000001784906045">
                <score xml:id="score-0000000289180743">
                    <scoreDef xml:id="scoredef-0000001786983899" midi.bpm="400">
                        <staffGrp xml:id="staffgrp-0000001354235601">
                            <staffDef xml:id="staffdef-0000001235850132" clef.shape="G" clef.line="2" meter.count="4" meter.unit="4" n="1" lines="5">
                                <label xml:id="label-0000000407058692" />
                            </staffDef>
                        </staffGrp>
                    </scoreDef>
                    <section xml:id="section-L1F1">
                        <measure xml:id="measure-L3" n="1">
                            <staff xml:id="staff-L3F1N1" n="1">
                                <layer xml:id="layer-L3F1N1" n="1">
                                    <note xml:id="note-L4F1" dur="2" oct="5" pname="c" stem.dir="up" accid.ges="n">
                                        <artic xml:id="artic-L4F1" artic="stacc" place="above" />
                                    </note>
                                    <note xml:id="note-L5F1" dur="2" oct="5" pname="d" stem.dir="up" accid.ges="n">
                                        <artic xml:id="artic-L5F1" artic="stacc" place="above" />
                                    </note>
                                </layer>
                            </staff>
                            <slur xml:id="slur-L4F1-L5F1N1" staff="1" startid="#note-L4F1" endid="#note-L5F1" curvedir="above" />
                        </measure>
                        <measure xml:id="measure-L6" right="end">
                            <staff xml:id="staff-L6F1N1" n="1">
                                <layer xml:id="layer-L6F1N1" n="1">
                                    <note xml:id="note-L7F1" dur="2" oct="5" pname="c" stem.dir="up" accid.ges="n">
                                        <artic xml:id="artic-L7F1" artic="acc" place="above" />
                                    </note>
                                    <note xml:id="note-L8F1" dur="2" oct="5" pname="d" stem.dir="up" accid.ges="n">
                                        <artic xml:id="artic-L8F1" artic="acc" place="above" />
                                    </note>
                                </layer>
                            </staff>
                            <slur xml:id="slur-L7F1-L8F1N1" staff="1" startid="#note-L7F1" endid="#note-L8F1" curvedir="above" />
                        </measure>
                    </section>
                </score>
            </mdiv>
        </body>
    </music>
</mei>
```

**Comment by donbyrd** [comment]
The enormous number of potential problem situations like this is what led me to argue in my dissertation that what's really needed is to do layout in _image space_, i.e., working with a graphical image of the music, not in _object space_. What are "situations like this"? They're kind of like "this sort of configuration", to use your words :-) . My comment probably isn't helpful now, but I think it's going to be clear at some point that Verovio's set of object-space layout rules is too complex to improve further. In my dissertation I suggested applying simulated annealing to an image-space representation to find an optimal layout; Chris Raphael pointed out how slow that's likely to be, and suggested quadratic programming instead.

**Comment by craigsapp** [comment]
Slur placement is complicated and difficult to quantify as well as anticipate all possibilities algorithmically.  When typesetting in SCORE, I spend the most time on slur placement than any other particular musical feature.

But in this particular case, these rules are symbolic in nature and precursors to any specific graphical layout.  For example, when a slur is requested to be placed above a chord, it should not be attached to the bottom note of a chord...

**Comment by craigsapp** [comment]
Here is another related example:

<img width="612" alt="screen shot 2018-07-01 at 3 33 33 pm" src="https://user-images.githubusercontent.com/3487289/42134942-93462a8c-7d44-11e8-88ce-62d1837674b5.png">

The slur is automatically oriented above the notes in both measures; however, in the first measure the slur is not displayed correctly because the slur is attached to the bottom note in the chord.  When a slur is placed above a chord, then it should be attached to the top note in the chord rather than to the head-note in the chord (the note which posses the stem).  For this example, the first slur should look exactly like the second one (or nearly so).

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="http://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="http://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="4.0.0">
    <meiHead>
        <fileDesc>
            <titleStmt>
                <title />
            </titleStmt>
            <pubStmt />
        </fileDesc>
        <encodingDesc>
            <appInfo>
                <application isodate="2018-07-01T15:33:19" version="2.0.0-dev-c25c14a">
                    <name>Verovio</name>
                    <p>Transcoded from Humdrum</p>
                </application>
            </appInfo>
        </encodingDesc>
        <workDesc>
            <work>
                <titleStmt>
                    <title />
                </titleStmt>
            </work>
        </workDesc>
    </meiHead>
    <music>
        <body>
            <mdiv xml:id="mdiv-0000000357619707">
                <score xml:id="score-0000000226923497">
                    <scoreDef xml:id="scoredef-0000001156821703">
                        <staffGrp xml:id="staffgrp-0000000218116419">
                            <staffDef xml:id="staffdef-0000000614312059" clef.shape="G" clef.line="2" meter.count="3" meter.unit="4" n="1" lines="5">
                                <label xml:id="label-0000001034683472" />
                            </staffDef>
                        </staffGrp>
                    </scoreDef>
                    <section xml:id="section-L1F1">
                        <measure xml:id="measure-L3" n="1">
                            <staff xml:id="staff-L3F1N1" n="1">
                                <layer xml:id="layer-L3F1N1" n="1">
                                    <rest xml:id="rest-L4F1" dur="8" />
                                    <chord xml:id="chord-L5F1" dur="4">
                                        <note xml:id="note-L5F1S1" oct="4" pname="b" accid.ges="n" />
                                        <note xml:id="note-L5F1S2" oct="5" pname="b" accid.ges="n" />
                                    </chord>
                                    <chord xml:id="chord-L6F1" dur="4">
                                        <note xml:id="note-L6F1S1" oct="4" pname="g" accid.ges="n" />
                                        <note xml:id="note-L6F1S2" oct="5" pname="g" accid.ges="n" />
                                    </chord>
                                    <chord xml:id="chord-L7F1" dur="8">
                                        <note xml:id="note-L7F1S1" oct="4" pname="e" accid.ges="n" />
                                        <note xml:id="note-L7F1S2" oct="5" pname="e" accid.ges="n" />
                                    </chord>
                                </layer>
                            </staff>
                            <slur xml:id="slur-L5F1-L7F1N1" staff="1" startid="#chord-L5F1" endid="#chord-L7F1" />
                        </measure>
                        <measure xml:id="measure-L8" n="2">
                            <staff xml:id="staff-L8F1N1" n="1">
                                <layer xml:id="layer-L8F1N1" n="1">
                                    <rest xml:id="rest-L9F1" dur="8" />
                                    <chord xml:id="chord-L10F1" dur="4">
                                        <note xml:id="note-L10F1S1" oct="4" pname="b" accid.ges="n" />
                                        <note xml:id="note-L10F1S2" oct="5" pname="b" accid.ges="n" />
                                    </chord>
                                    <chord xml:id="chord-L11F1" dur="4">
                                        <note xml:id="note-L11F1S1" oct="4" pname="g" accid.ges="n" />
                                        <note xml:id="note-L11F1S2" oct="5" pname="g" accid.ges="n" />
                                    </chord>
                                    <chord xml:id="chord-L12F1" dur="8" stem.dir="down">
                                        <note xml:id="note-L12F1S1" oct="4" pname="e" accid.ges="n" />
                                        <note xml:id="note-L12F1S2" oct="5" pname="e" accid.ges="n" />
                                    </chord>
                                </layer>
                            </staff>
                            <slur xml:id="slur-L10F1-L12F1N1" staff="1" startid="#chord-L10F1" endid="#chord-L12F1" />
                        </measure>
                    </section>
                </score>
            </mdiv>
        </body>
    </music>
</mei>
```

**Comment by craigsapp** [comment]
And here is another case of the same problem as in the last example:

<img width="577" alt="screen shot 2018-07-01 at 3 54 31 pm" src="https://user-images.githubusercontent.com/3487289/42135078-90a80b7c-7d46-11e8-8899-deed57370419.png">

The slur is placed above the chords, but the slur is attached to the head-notes of the chord rather than the top notes.

MEI test data:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="http://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="http://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="4.0.0">
    <meiHead>
        <fileDesc>
            <titleStmt>
                <title />
            </titleStmt>
            <pubStmt />
        </fileDesc>
        <encodingDesc>
            <appInfo>
                <application isodate="2018-07-01T15:54:28" version="2.0.0-dev-c25c14a">
                    <name>Verovio</name>
                    <p>Transcoded from Humdrum</p>
                </application>
            </appInfo>
        </encodingDesc>
        <workDesc>
            <work>
                <titleStmt>
                    <title />
                </titleStmt>
            </work>
        </workDesc>
        <extMeta>
            <frames xmlns:humxml="http://www.humdrum.org/ns/humxml">
                <metaFrame n="11" token="!!!RDF**kern: &lt; = below" xml:id="loc11">
                    <frameInfo>
                        <startTime float="8" />
                        <frameType>reference</frameType>
                        <referenceKey>RDF**kern</referenceKey>
                        <referenceValue>&lt; = below</referenceValue>
                    </frameInfo>
                </metaFrame>
                <metaFrame n="12" token="!!!RDF**kern: &gt; = above" xml:id="loc12">
                    <frameInfo>
                        <startTime float="8" />
                        <frameType>reference</frameType>
                        <referenceKey>RDF**kern</referenceKey>
                        <referenceValue>&gt; = above</referenceValue>
                    </frameInfo>
                </metaFrame>
            </frames>
        </extMeta>
    </meiHead>
    <music>
        <body>
            <mdiv xml:id="mdiv-0000000891391643">
                <score xml:id="score-0000000951853025">
                    <scoreDef xml:id="scoredef-0000000401173812" midi.bpm="400">
                        <staffGrp xml:id="staffgrp-0000001169263445" symbol="brace" barthru="true">
                            <staffDef xml:id="staffdef-0000000303009138" clef.shape="G" clef.line="2" meter.count="4" meter.unit="4" n="1" lines="5">
                                <label xml:id="label-0000000779169277" />
                            </staffDef>
                            <staffDef xml:id="staffdef-0000001174712668" clef.shape="F" clef.line="4" meter.count="4" meter.unit="4" n="2" lines="5" />
                        </staffGrp>
                    </scoreDef>
                    <section xml:id="section-L1F1">
                        <measure xml:id="measure-L4">
                            <staff xml:id="staff-L4F2N1" n="1">
                                <layer xml:id="layer-L4F2N1" n="1">
                                    <chord xml:id="chord-L5F2" dur="2">
                                        <note xml:id="note-L5F2S1" staff="2" oct="3" pname="e" accid.ges="n" />
                                        <note xml:id="note-L5F2S2" oct="4" pname="e" accid.ges="n" />
                                    </chord>
                                    <chord xml:id="chord-L6F2" dur="2">
                                        <note xml:id="note-L6F2S1" staff="2" oct="3" pname="c" accid.ges="n" />
                                        <note xml:id="note-L6F2S2" oct="4" pname="c" accid.ges="n" />
                                    </chord>
                                </layer>
                            </staff>
                            <staff xml:id="staff-L4F1N1" n="2">
                                <layer xml:id="layer-L4F1N1" n="1">
                                    <mRest xml:id="mrest-L5F1" visible="false" />
                                </layer>
                            </staff>
                            <slur xml:id="slur-L5F2-L6F2N1" staff="1" startid="#chord-L5F2" endid="#chord-L6F2" curvedir="above" />
                        </measure>
                        <measure xml:id="measure-L7">
                            <staff xml:id="staff-L7F2N1" n="1">
                                <layer xml:id="layer-L7F2N1" n="1">
                                    <chord xml:id="chord-L8F2" dur="2">
                                        <note xml:id="note-L8F2S1" oct="4" pname="e" accid.ges="n" />
                                        <note xml:id="note-L8F2S2" oct="5" pname="e" accid.ges="n" />
                                    </chord>
                                    <chord xml:id="chord-L9F2" dur="2">
                                        <note xml:id="note-L9F2S1" oct="4" pname="c" accid.ges="n" />
                                        <note xml:id="note-L9F2S2" oct="5" pname="c" accid.ges="n" />
                                    </chord>
                                </layer>
                            </staff>
                            <staff xml:id="staff-L7F1N1" n="2">
                                <layer xml:id="layer-L7F1N1" n="1">
                                    <mRest xml:id="mrest-L8F1" visible="false" />
                                </layer>
                            </staff>
                            <slur xml:id="slur-L8F2-L9F2N1" staff="1" startid="#chord-L8F2" endid="#chord-L9F2" curvedir="above" />
                        </measure>
                    </section>
                </score>
            </mdiv>
        </body>
    </music>
</mei>
```

**Comment by donbyrd** [comment]
You're right, it's very clear what to do in these cases, specifically to attach the slur to the upper note of the last chord. But make the situation more complex -- for example, with a 2nd voice between the notes of the existing voice, which would not be too unusual in piano music -- and it'd be different story.

**Comment by craigsapp** [comment]
The last case was distilled from this real-world case:

<img width="257" alt="screen shot 2018-07-01 at 9 25 40 pm" src="https://user-images.githubusercontent.com/3487289/42137919-0afc9676-7d75-11e8-8035-b1856de8fa38.png">

This gets pretty complicated, but the basic "above" description would be sufficient in this case.  If the inner voice were with stems up, or the pitch of the inner voice were higher, then the slur layout would get more complex.

**Comment by lpugin** [comment]
Fixed in 3.2.0-dev-9f116da

![image](https://user-images.githubusercontent.com/689412/105143941-d19e4100-5afc-11eb-8da2-2d84a81c64d4.png)

![image](https://user-images.githubusercontent.com/689412/105144091-027e7600-5afd-11eb-8737-11d553792bd9.png)
